Add CVE justification for CVE-2022-45868 (#69)
wso2/docs-security: en/docs/security-announcements/cve-justifications/2024/CVE-2022-45868.md

---
title: "CVE-2022-45868"
category: security-announcements
date: July 24 2024
---

# CVE-2022-22965

<p class="doc-info">WSO2 Products impacted: no</p>
<p class="doc-info">Customers actions required: no</p>
---

### REPORTED VULNERABILITY

> The web-based admin console in H2 Database Engine before 2.2.220 can be started via the CLI with the argument -webAdminPassword, which allows the user to specify the password in cleartext for the web admin console. Therefore, a local user is able to access the password by listing the processes and their arguments [^1].

Reviewer: ```suggestion
The web-based admin console in H2 Database Engine before 2.2.220 can be started via the CLI with the argument `-webAdminPassword`, which allows the user to specify the password in cleartext for the web admin console. Therefore, a local user is able to access the password by listing the processes and their arguments [^1].
```

### REPORTED PRODUCTS
* WSO2 API Manager : 4.3.0, 4.2.0, 4.1.0, 4.0.0, 3.2.0, 3.1.0, 3.0.0

> * WSO2 Open Banking AM : 2.0.0, 1.5.0, 1.4.0

Reviewer 2: ```suggestion
* WSO2 Open Banking AM : 2.0.0, 1.5.0, 1.4.0
* Any other WSO2 products containing the H2 database engine 
```

### WSO2 JUSTIFICATION

> In WSO2 API Manager the H2 web-based console is disabled by default and the deployment.toml configuration related feature to enable/disable the H2 web console is removed from APIM 4.2.0 onwards. Moreover, the vulnerability is reported when starting the H2 web based console using the CLI with a clear text password which is not applicable for API Manager.

Reviewer 2: ```suggestion
The vulnerability is reported when starting the H2 web based console using the CLI with a clear text password which is not applicable for WSO2 products.

Furthermore, the H2 web based console is disabled by default, and newer WSO2 product versions including APIM 4.2.0 onwards even removes deployment.toml configuration option to enable/disable the H2 web console. 
```

> Furthermore, the CVE-2022-45868 is a disputed vulnerability since the vendor states that "This is not a vulnerability of H2 Console Passwords should never be passed on the command line and every qualified DBA or system administrator is expected to know that."

Reviewer: ```suggestion
Furthermore, the CVE-2022-45868 is a disputed vulnerability since the vendor states that, 
> This is not a vulnerability of H2 Console Passwords should never be passed on the command line and every qualified DBA or system administrator is expected to know that.
```

---
title: "CVE-2022-45868"
category: security-announcements
date: July 24 2024
---

# CVE-2022-22965

<p class="doc-info">WSO2 Products impacted: no</p>
<p class="doc-info">Customers actions required: no</p>
---

### REPORTED VULNERABILITY
The web-based admin console in H2 Database Engine before 2.2.220 can be started via the CLI with the argument `-webAdminPassword`, which allows the user to specify the password in cleartext for the web admin console. Therefore, a local user is able to access the password by listing the processes and their arguments [^1].

### REPORTED PRODUCTS
* WSO2 API Manager : 4.3.0, 4.2.0, 4.1.0, 4.0.0, 3.2.0, 3.1.0, 3.0.0
* WSO2 Open Banking AM : 2.0.0, 1.5.0, 1.4.0

### WSO2 JUSTIFICATION
In WSO2 API Manager the H2 web-based console is disabled by default and the deployment.toml configuration related feature to enable/disable the H2 web console is removed from APIM 4.2.0 onwards. Moreover, the vulnerability is reported when starting the H2 web based console using the CLI with a clear text password which is not applicable for API Manager.

> Furthermore, the CVE-2022-45868 is a disputed vulnerability since the vendor states that, 

Reviewer 2: ```suggestion
Furthermore, the [CVE-2022-45868](https://nvd.nist.gov/vuln/detail/CVE-2022-45868) is a disputed vulnerability and the project states the following: 
```

Reviewer: ```suggestion
Furthermore, the CVE-2022-45868[^1] is a disputed vulnerability and the project states the following: 
```

> This is not a vulnerability of H2 Console Passwords should never be passed on the command line and every qualified DBA or system administrator is expected to know that.

### CONCLUSION
Due to following key points,

> * [CVE-2022-45868](https://nvd.nist.gov/vuln/detail/CVE-2022-45868) is reported for H2 database engine before 2.2.220

Reviewer 2: ```suggestion
* [CVE-2022-45868](https://nvd.nist.gov/vuln/detail/CVE-2022-45868) is a disputed CVE.
```

Reviewer: ```suggestion
* CVE-2022-45868[^1] is a disputed CVE.
```

> * H2 console has been disabled by default and the configuration to enable the console is removed from APIM 4.2.0 onwards.

Reviewer 2: ```suggestion
* H2 console has been disabled by default in WSO2 products.
* Vulnerability is reported when starting the H2 web based console using the CLI which is never applicable to WSO2 products.
* Upgrading to a H2 version that is not flagged for this vulnerability requires backward incompatible changes to the product. This requires users of WSO2 products to perform a data migration, when the vulnerability itself can never affect security of WSO2 products.
```

> Therefore, this is not an exploitable vulnerability in WSO2 products.

Reviewer 2: ```suggestion
Therefore, WSO2 concludes that this is not an exploitable vulnerability in WSO2 products, and an H2 update will not be performed due to [CVE-2022-45868](https://nvd.nist.gov/vuln/detail/CVE-2022-45868).
```

Reviewer: ```suggestion
Therefore, WSO2 concludes that **this is not an exploitable vulnerability in WSO2 products**, and **an H2 update will not be performed due to CVE-2022-45868**[^1].
```
